client/models/scenery_editor.js

```diff
@@ -318,16 +321,76 @@ class SceneryEditor {
             post('api/tools/scenery/instance/delete/' + WORKSPACES.opened, {
                 id: document.getElementById('tools-detail-scenery-id').innerText
             }, () => {
-                WORKSPACES.reload();
+                this.removeObject(document.getElementById('tools-detail-scenery-id').innerText);
             });
         } else if (this.selected_type === 'unique') {
             post('api/tools/scenery/unique/delete/' + WORKSPACES.opened, {
                 id: document.getElementById('tools-detail-scenery-id').innerText
             }, () => {
-                WORKSPACES.reload();
+                this.removeUnique(document.getElementById('tools-detail-scenery-id').innerText);
             });
         }
     }
+
+    addObject(object){
+        let object_id = object.x + ',' + object.y;
+            
+        WORKSPACES.current_map.sceneryLoader.createScenery(object, (model, definition) => {
+            let m = createSceneryMesh(object_id, object, WORKSPACES.current_map.terrain, model, definition);
+            m.original_id = { type: 'scenery', id: object_id };
+            WORKSPACES.current_map.scenery_groups['trees'].add(m);
+            WORKSPACES.current_map.scenery_references[object_id] = {
+                instance: object,
+                definition: definition,
+                threeObject: m
+            }
+        })
+    }
+
+    editObjectTint(object){
+        let newObject = WORKSPACES.current_map.scenery_references[object.id].instance;
```

Comment:
This doesn't actually create a new object. In this case, it's not a problem, but something to keep in mind -- if you weren't deleting this before, newObject.foo = 3 would actually change the **old** object as well!